tests/base/deterministic_model.py

```diff
@@ -162,13 +166,61 @@ def training_epoch_end_dict(self, outputs):
 
         return {'log': logs, 'progress_bar': pbar}
 
+    def validation_step_no_return(self, batch, batch_idx):
+        self.validation_step_called = True
+        acc = self.step(batch, batch_idx)
+
+    def validation_step_scalar_return(self, batch, batch_idx):
+        self.validation_step_called = True
+        acc = self.step(batch, batch_idx)
+        return acc
+
+    def validation_step_arbitary_dict_return(self, batch, batch_idx):
+        self.validation_step_called = True
+        acc = self.step(batch, batch_idx)
+        return {'some': acc, 'value': 'a'}
+
     def validation_step_dict_return(self, batch, batch_idx):
+        self.validation_step_called = True
         acc = self.step(batch, batch_idx)
 
-        logs = {'log_acc1': torch.tensor(12).type_as(acc), 'log_acc2': torch.tensor(7).type_as(acc)}
+        logs = {'log_acc1': torch.tensor(12 + batch_idx).type_as(acc), 'log_acc2': torch.tensor(7).type_as(acc)}
         pbar = {'pbar_acc1': torch.tensor(17).type_as(acc), 'pbar_acc2': torch.tensor(19).type_as(acc)}
         return {'val_loss': acc, 'log': logs, 'progress_bar': pbar}
 
+    def validation_step_end_no_return(self, val_step_output):
+        assert len(val_step_output) == 3
+        assert val_step_output['val_loss'] == 171
```

Comment:
constants like 171 which are used to assert results in tests should be defined as constants in the deterministic_model.py.

ex- CONST_1 = 171
and then from tests.base import CONST_1
and then assert

Reply:
but these change a bit depending on the particular setup.
I agree that we can make them constants later, but for now i'd rather be able to iterate through the tests faster.